src/ld/OutputFile.cpp

```diff
@@ -4575,6 +4575,10 @@ int OutputFile::compressedOrdinalForAtom(const ld::Atom* target)
 	// regular ordinal
 	const ld::dylib::File* dylib = dynamic_cast<const ld::dylib::File*>(target->file());
 	if ( dylib != NULL ) {
+		// Handle -dynamic_lookup_library
+		if ( dylib->forcedDynamicLookupLinked() )
+			return BIND_SPECIAL_DYLIB_FLAT_LOOKUP;
```

Comment:
There is also `BIND_SPECIAL_DYLIB_MAIN_EXECUTABLE`. However, I suspect this is more geared towards the case where the symbols are actually *in* the main executable which is not the case for the `libpython` use case.